fw/http.c

```diff
@@ -5417,6 +5444,11 @@ tfw_http_req_process(TfwConn *conn, TfwStream *stream, const TfwFsmData *data)
 					     " processing error");
 		TFW_INC_STAT_BH(clnt.msgs_otherr);
 	}
+	else if (unlikely(req->resp) && unlikely(req->cache_ctl.flags &
+		TFW_HTTP_CC_CACHE_PURGE)) {
+		tfw_http_send_resp(req, 200, "purge: success");
```

Comment:
Well, the design of the pull request isn't so far from the discussed one. This `tfw_http_send_resp(req, 200...)` call is very confusing though. There are also several issues with the caching hooks and calls (also commented previously) and the HTTP requests adjustments. Fixing all the issues will lead to the desired changes.